### ISSUE TITLE
Cross file template type checking - check that components are passed all declared props

### ISSUE THREAD

**Issue by octref** [vuejs/vetur#2135]
Needed for #1596.

Assume `Child.vue`,

```vue
<script>
export default {
  props: ['foo', 'bar']
}
</script>
```

and `Parent.vue`,

```vue
<script>
import Child from './Child.vue'

export default {
  components: {
    Child
  },
  props: ['foo', 'bar']
}
</script>
```

In `Parent.vue`, these diagnostics should be shown:

```vue
<template>
  <!--Missing `foo` and `bar` -->
  <child></child>

  <!--Missing `bar` -->
  <child :foo="foo"></child>

  <!--Missing `bar` -->
  <!--`baz` is not defined in `Child.vue` -->
  <child :foo="foo" :baz="bar"></child>
</template>
```

**Comment by yoyo930021** [comment]
Sometimes we don't need to have prop
When
```typescript=
export default {
  props: {
    active: {
      type: Boolean,
      default: true
    }
  }
}
```

**Comment by octref** [comment]
@yoyo930021 Good catch!

**Comment by yoyo930021** [comment]
> @yoyo930021 Good catch!
> 
> Update: Are you sure? This seems to work fine for me:
> 
> Showing diagnostics:
> 
> ![image](https://user-images.githubusercontent.com/4033249/89705397-9d402980-d98f-11ea-8872-e70691479464.png)
> 
> Showing no diagnostics:
> 
> ![image](https://user-images.githubusercontent.com/4033249/89705405-ab8e4580-d98f-11ea-80de-2e7e8badcb64.png)

When I use default in prop,
I don't use prop in template.
Like ensure compatibility

**Comment by octref** [comment]
I'll fix it in #2141.
